src/cobra/core/compartment.py

```diff
@@ -180,25 +123,64 @@ def metabolites(self) -> DictList[Metabolite]:
         return self._members.query(lambda x: isinstance(x, Metabolite))
 
     @property
-    def reactions(self) -> Optional[FrozenSet[Reaction]]:
+    def inferred_reactions(self) -> FrozenSet[Reaction]:
         """Return the reactions whose metabolites are in the compartment.
 
         This is returned as a FrozenSet of reactions for each metabolite in the
         compartment, if any.
 
+        Returns
+        -------
+        FrozenSet of cobra.Reactions
+            Reactions that have metabolites that belong to this compartment.
+
         Returns
         -------
         FrozenSet of cobra.Reactions
             Reactions that have metabolites that belong to this compartment.
         """
-        direct_set = set(self._members.query(lambda x: isinstance(x, Reaction)))
         rxn_set = set()
         for met in self.metabolites:
             rxn_set.update(met._reactions)
-        return frozenset(rxn_set.union(direct_set))
+        return frozenset(rxn_set)
+
+    @property
+    def assigned_reactions(self) -> FrozenSet[Reaction]:
+        """Return the reactions who were assigned to this compartment.
+
+        This is returned as a FrozenSet of reactions for each metabolite in the
+        compartment, if any.
+
+        Returns
+        -------
+        FrozenSet of cobra.Reactions
+            Reactions that have metabolites that belong to this compartment.
+
+        Returns
+        -------
+        FrozenSet of cobra.Reactions
+            Reactions that were assigned to this compartment, if any.
+        """
+        return frozenset(self._members.query(lambda x: isinstance(x, Reaction)))
+
+    @property
+    def reactions(self) -> FrozenSet[Reaction]:
+        """Return the reactions who belong to this compartment.
+
+        This is returned as a FrozenSet of reactions for each metabolite in the
+        compartment, if any, and the reactions that were assigned to this compartment
+        directly.
+
+        Returns
+        -------
+        FrozenSet of cobra.Reactions
+            Reactions that belong to this compartment, both assigned and inferred.
+        """
+        assigned_reactions = set(self.assigned_reactions)
+        return frozenset(assigned_reactions.union(self.inferred_reactions))
 
-    def __contains__(self, member: Union[Metabolite, Gene]):
-        return member.compartment is self
+    def __contains__(self, member: Union[Metabolite, Reaction]):
+        return self.members.__contains__(member)
```

Comment:
If we keep members then more pythonic would be
```suggestion
        return member in self.members
```